packages/context/src/binding.ts

```diff
@@ -251,6 +306,36 @@ export class Binding {
     return this;
   }
 
+  /**
+   * Set options for the binding
+   * @param options Options object or a promise
+   */
+  withOptions(options: BindingOptions): this {
```

Comment:
Following up on https://github.com/strongloop/loopback-next/pull/671#discussion_r160131295, I think this is the only place where we need to deal with `RejectionError` stuff and convert a promise of `options` into a getter function, allowing us to retain control of when we want the rejection error to (re)surface. (Ignoring the usability problem of delaying app-setup errors.)